### PR TITLE
WIP: Clean geometry

### PR DESCRIPTION
**Pull request by chrlecarpentier**
Removed dispensable shapes (null length internodes, CutPointBud cylinders and inside sheath) to simplify plants representation

### CODE REVIEW

**Comment by christian34** [comment]
@chrlecarpentier I have change the geometry to get visible length: can you check the result is Okay for you ? (if you want to test within lpy, may be you can merge #16 in master then in branch before)

**Comment by pradal** [comment]
@christian34 I have merged the PR #16, do not worry.

**Comment by christian34** [comment]
@chrlecarpentier @Emblanc In fact you have the choice between two geometry for the stem:
- the one proposed by Christophe (accessible by reverting this branch to commit 
df87325 ) with one cylinder for the whole stem, and keepiing the topology as in master ( [Bud] Internode [Sheath Blade].
- the one I implement, with one cylinder per visible organ (internode or sheath), but with a change of topology of the string : [Bud] Internode Sheath [Blade]. The later is closer to the one I use in adel.

The main difference is for PAR aggregation: with solution 1 you have one PAR value for the whole stem, with solution 2 you have 1 PAR per sheath/internode

**Comment by christian34** [comment]
Yes sure !

**Comment by christian34** [comment]
Master has been meged again: I will add some lines to answer issue #47

**Comment by Emblanc** [comment]
I have started testing this new branch and there seems to be at least one problem. In this PR, some organs that should be represented are no longer represented. Plants now have "flying ears".
mastervscleangeometry

**Comment by christian34** [comment]
Yes : the branch is still in progress: I will post a comment when the geometry is...clean again !

**Comment by christian34** [comment]
@Emblanc Okay the branch is ready to test ! Using actual primitive area instead of visibility estimate alows to stabilise a bit more the max PAR variation.

**Comment by Emblanc** [comment]
I tested this branch and it seems to be working fine. However, there is something odd that I do not understand.
When I compare the results of a simulation in the clean_geometry branch and in master, here is what I get :

1) Most of the time (for more than 87% of the cases) : sum_primitive > tiller_surface (this is OK for me)
2) Sum_PAR master > Sum_PAR Clean_geometry.
This is consistent with 1) since SumPAR = interceptedPAR/surface and interceptedPAR is the same for master and Clean_geometry (very little variation)
3) There are more regressing tillers in master than in Clean_geometry (this is not consistent with 2)). Normally, the more light the plot intercepts, the less tillers regress...

Apart from this behaviour (3)) that I do not understand, I did not see any problem with the branch.

File: src/walter_data/WALTer.lpy [hunk 1]
```diff
@@ -2347,17 +2351,22 @@ def EndEach(lstring, lscene):
...
-                PAR_per_axes_dico["Temp_cum"].append(Tempcum)
-                PAR_per_axes_dico["Num_plante"].append(num_plt)
-                PAR_per_axes_dico["Num_talle"].append(axis)
-                PAR_per_axes_dico["Inc_PAR"].append(current_PAR)
-                tt = round(Tempcum, 1)
-                PAR_per_axes_dico["Sum_PAR"].append(dico_PAR_per_axis[num_plt][axis].get(tt,0.))
-                PAR_per_axes_dico["Abs_int_PAR"].append(dico_absolute_PAR_per_axis[num_plt][axis].get(tt,0.))
+            par_per_axe = pd.DataFrame(PAR_per_organ).groupby(['Num_plante', 'Num_talle']).agg({'Organ_PAR': 'sum','Organ_surface': 'sum', 'tiller_surface':'mean'})
+            for (nump, numt), row in par_per_axe.iterrows():
+              dico_PAR_per_axis[nump][numt][round(Tempcum, 1)] = row.Organ_PAR / Temperature / row.Organ_surface
```
Comment (pradal) [inline src/walter_data/WALTer.lpy]: This line is not clear for me. Does that mean :
row.Organ_PAR * row.Organ_surface / Temperature

Comment (Emblanc) [reply src/walter_data/WALTer.lpy]: To me, it means : (row.Organ_PAR / Temperature) / row.Organ_surface ; so the result is in µmolPAR.degree_day-¹.cm-²

**Comment by christian34** [comment]
@Emblanc I remove merge conflict and all test pass: the branch is ready to be merged for me !
(I will base on this one for testing curved leaves)

**Comment by Emblanc** [comment]
I made a few changes to solve issues #65 #67 #66

**Pull request merged**